### PR TITLE
[stable/filebeat] set default config empty

### PR DESCRIPTION
**Pull request by kimxogus**
Signed-off-by: Taehyun Kim <kgyoo8232@gmail.com>

#### What this PR does / why we need it:
#12199

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes filebeat's #12199 issue

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by sstarcher** [comment]
So by default this will do pretty much nothing.  Any reason you could not override the values that you want?  I don't believe setting the default setting to be useless is a good move and we should consider something else.

**Comment by cdenneen** [comment]
https://github.com/helm/helm/issues/5184 if working properly means you can override the `output.file` to `null` "remove" it but it doesn't seem to be working properly. If this worked properly you most likely wouldn't need to have empty configuration?

**Comment by kimxogus** [comment]
@sstarcher `prospector` was deprecated since 6.3, and deprecation warning log is printed in this chart.
Setting `filebeat.prospectors` as `null` or empty list `[]` couldn't unset prospectors, so `filebeat.inputs` can't be used.

**Comment by sstarcher** [comment]
That's fine so this configuration should take a different tact.  We should not wipe out the default config.

Possibly a new option
```
overrideConfig: |
  Whatever random config you want
```

If `overrideConfig` is set use it and don't use the default at all.

My concern with these changes is you are breaking everyone and the default configuration for this chart is basically useless.  Why don't we try and find a way to make both a reality.

**Comment by cdenneen** [comment]
@kimxogus when the bug for “null” works properly what issue will that not fix?

**Comment by kimxogus** [comment]
@sstarcher `overrideConfig` seems like a good solution :)
Added `overrideConfig` as suggested.

**Comment by kimxogus** [comment]
merged master

**Comment by kimxogus** [comment]
fixed conflicts. @sstarcher can you please review?

File: stable/filebeat/templates/daemonset.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ spec:
         app: {{ template "filebeat.name" . }}
         release: {{ .Release.Name }}
       annotations:
-        checksum/secret: {{ toYaml .Values.config | sha256sum }}
+        checksum/secret: {{ toYaml (default .Values.config .Values.overrideConfig) | sha256sum }}
```
Comment (sstarcher) [inline stable/filebeat/templates/daemonset.yaml:26]: going to approve, but I believe it makes it easier to read if written
`        checksum/secret: {{ .Values.overrideConfig | default .Values.config | toYaml | sha256sum }}`

**Pull request merged**